packages/skeleton-svelte/src/components/Tabs/TabsControl.svelte

```diff
@@ -19,33 +22,59 @@
 		// Snippets
 		lead,
 		children,
-		// Zag
+		// Zag props
 		...zagProps
 	}: TabsControlProps = $props();
 
-	// Get Context
+	// Context
 	const ctx = getTabContext();
 	const state = $derived(ctx.api.getTriggerState(zagProps));
 
-	// Reactive
+	// State classes
 	const rxActive = $derived(state.selected ? stateActive : stateInactive);
 	const rxLabelActive = $derived(state.selected ? stateLabelActive : stateLabelInactive);
 
-	// Styles
+	// Layout
 	const commonWidth = $derived(ctx.fluid ? '100%' : '');
+
+	// Get all trigger props from Zag.js
+	const triggerProps = ctx.api.getTriggerProps(zagProps);
 </script>
 
-<!-- @component A individual tab trigger element. -->
-
-<button
-	{...ctx.api.getTriggerProps(zagProps)}
-	class="{base} {padding} {translateX} {rxActive} {classes}"
-	style:width={commonWidth}
-	data-testid="tabs-control"
->
-	<!-- Label -->
-	<div class="{labelBase} {rxLabelActive} {labelClasses}" style:width={commonWidth} data-testid="tabs-control-label">
-		{#if lead}<span>{@render lead()}</span>{/if}
-		<span>{@render children?.()}</span>
-	</div>
-</button>
+<!-- @component Individual tab trigger (anchor or button) -->
+
+{#if as === 'a'}
```

Comment:
No need for duplication of code like this, as Svelte provides the `<svelte:element>` to solve for dynamic element types. See the docs here for details on the usage:
https://svelte.dev/docs/svelte/svelte-element

React offers a similar solution. Just note the use of capital letters for defining the dynamic element:
https://stackoverflow.com/a/33471928